xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/EntityReference.java

```diff
@@ -679,6 +687,31 @@ public boolean equalsNonRecursive(EntityReference otherReference)
             && (parameters == null ? otherReference.parameters == null : parameters.equals(otherReference.parameters));
     }
 
+    /**
+     * The parent type information is used by resolvers to identify which part of the base reference should be kept.
+     * If the entity reference has a parent (see {@link #getParent()}) then this type should always be the type of
+     * the parent. Now if the entity reference doesn't have the parent this value can be given by the
+     * {@link #PARENT_TYPE_PARAMETER} parameter (see {@link #getParameter(String)}), and if none is given it will
+     * fall back on first allowed parents (see {@link EntityType#getAllowedParents()} of current type returned by
+     * {@link #getType()}.
+     * @return the type of the parent to be used for computing the proper base reference in resolvers.
+     * @since 17.0.0RC1
+     */
+    @Unstable
+    public EntityType getParentType()
+    {
+        EntityType result;
+        if (parent == null) {
```

Comment:
Minor codestyle: since you are using `getType()` it would probably make more sense to use `getParent()`, or use `this.type` instead of `getType()`.

Reply:
indeed I'm not very consistent here.